service/frontend/workflow_handler.go

```diff
@@ -3707,16 +3708,45 @@ func (wh *WorkflowHandler) StartBatchOperation(
 	}
 
 	// Validate concurrent batch operation
+	maxConcurrentBatchOperation := wh.config.MaxConcurrentBatchOperation(request.GetNamespace())
 	countResp, err := wh.CountWorkflowExecutions(ctx, &workflowservice.CountWorkflowExecutionsRequest{
 		Namespace: request.GetNamespace(),
 		Query:     batcher.OpenBatchOperationQuery,
 	})
-	if err != nil {
+	if err == nil {
```

Comment:
nit: how about refactoring it this way: 
```
if err != nil {
    if !errors.Is(err, store.OperationNotSupportedErr) {
        return nil, err
    }
    countResp = 0 
    nextPageToken := []byte{}
    for nextPageToken != nil && countResp < maxConcurrentBatchOperation {
			listResp, err := wh.ListOpenWorkflowExecutions(ctx, &workflowservice.ListOpenWorkflowExecutionsRequest{
				Namespace: request.GetNamespace(),
				Filters: &workflowservice.ListOpenWorkflowExecutionsRequest_TypeFilter{
					TypeFilter: &filterpb.WorkflowTypeFilter{
						Name: batcher.BatchWFTypeName,
					},
				},
				MaximumPageSize: int32(maxConcurrentBatchOperation - countResp),
				NextPageToken:   nextPageToken,
			})
			if err != nil {
				return nil, err
			}
			countResp += len(listResp.Executions)
			nextPageToken = listResp.NextPageToken
		}
}

if countResp.GetCount() >= int64(maxConcurrentBatchOperation) {
    return nil, serviceerror.NewResourceExhausted(enumspb.RESOURCE_EXHAUSTED_CAUSE_CONCURRENT_LIMIT, "Max concurrent batch operations is reached")
```
just a bit more compact, and ideally we change `countResp` to `countOpenWorkflows` or sth like that.

Reply:
Not exactly possible the way you proposed, but I get your point. Done.